### PR TITLE
Respect original logger level if not overridden

### PR DESCRIPTION
**Pull request by adamlogic**
When the original logger is set to INFO, we should only write DEBUG logs if the `log_level` config has been explicitly set for the adapter.

### CODE REVIEW

File: judoscale-ruby/test/logger_test.rb [hunk 1]
```diff
@@ -64,8 +64,16 @@ module Judoscale
...
+
+        use_config log_level: :info do
+          logger.debug "some message"
+          _(messages).wont_include "INFO -- : [Judoscale] [DEBUG] some message"
+        end
+
+        use_config log_level: :debug do
+          logger.debug "some message"
+          _(messages).must_include "INFO -- : [Judoscale] [DEBUG] some message"
+        end
```
Comment (adamlogic) [inline judoscale-ruby/test/logger_test.rb:76]: @carlosantoniodasilva I wrote this failing test for how I was expecting this to work. If this makes sense to you, would you mind taking over this PR to get the test passing?

Comment (carlosantoniodasilva) [reply judoscale-ruby/test/logger_test.rb:78]: @adamlogic ok, I think I got it right this time:

https://user-images.githubusercontent.com/26328/159821382-9c136acc-afdc-40ac-8865-b90985fd8605.mp4

From the commit:

```
If you don't set a log_level config, we simply want to delegate down to
the underlying logger to let it decide whether or not we should log.
This way, if you set the app's logger to INFO, all Judoscale DEBUG logs
won't show up by default, without any extra config.

If you explicitly set a log_level of INFO or above, we won't log
anything on DEBUG.

If you do set the log_level to DEBUG, we will log using the underlying
logger's log level, to ensure that Judoscale DEBUG logs are always there
no matter which log level the underlying logger is set to. This allows
us to log Judoscale DEBUG messages temporarily even if the underlying
logger is set to INFO or other for example.
```

I seem to remember I had a similar branch when I first did the log changes in https://github.com/judoscale/judoscale-ruby/pull/37, but ended up collapsing into that single line that is there now, thinking it was covering all cases that way, turns out the most important one wasn't covered. 😅

File: judoscale-ruby/lib/judoscale/logger.rb [hunk 2]
```diff
@@ -19,14 +19,20 @@ class LoggerProxy < Struct.new(:logger, :log_level)
 
       define_method(severity_name.downcase) do |*messages|
         if log?(severity_level)
-          logger.add(severity_level, tag(messages))
+          logger.add(severity_level) { tag(messages) }
```
Comment (carlosantoniodasilva) [inline judoscale-ruby/lib/judoscale/logger.rb:22]: I pushed a second commit moving these to blocks, it avoids the log messages from being processed (i.e. no tag string concatenation) if the logs aren't going to be really printed, which should save some work.

**Comment by adamlogic** [comment]
I'll go ahead and merge this PR since I created it. We can circle back if we want to make any more changes.

**Comment by carlosantoniodasilva** [comment]
Cool, thanks for the merge. I actually ran into a similar scenario yesterday, right after the video I recorded I tried to set the Rails logger level higher and Judoscale level to INFO, and indeed no Judoscale logs were output, which I guess is expected from the implementation.

I think it'd make sense to have the same implementation across all levels, so you could enforce any level to Judoscale regardless of the underlying logger level. The one thing that maybe would bother me a bit would be to prefix the severity in some cases, but not in others, mainly from the "searching the logs" perspective. Right now DEBUG logs are always prefixed with that, no matter which level the underlying logger is in, we did that to sort of normalize the DEBUG one. If we were to extend the implementation to others, we wouldn't want to prefix everything with the severity as you detailed in pseudocode too, but then we probably shouldn't do it anymore for DEBUG either (if the underlying logger is also DEBUG). The end result is perhaps a bit of inconsistency of what we log (or how the log is formatted), depending on the severity of the underlying logger.

Let's move this back to the drawing board for a bit as we consider making more changes.

**Pull request merged**